src/rpc/mining.cpp

```diff
@@ -729,6 +744,9 @@ UniValue getblocktemplate(const UniValue& params, bool fHelp)
     result.pushKV("bits", strprintf("%08x", pblock->nBits));
     result.pushKV("height", (int64_t)(pindexPrev->nHeight+1));
 
+    // Be sure not to return a cached coinbase-only block on the next call to this rpc
+    if (next_coinbase_mtx) pindexPrev = nullptr;
```

Comment:
This (discarding the cached block template) probably isn't needed, because the cached block template expires after 5 seconds, and the caller used `longpollid` on this call (or else `next_coinbase_mtx` wouldn't be valid here), so the caller will probably use `longpollid` on the next call, and then we'll block for at least 10 seconds (or until a new block arrives, which discards the cache). But this seems slightly safer.